cli/lib/kontena/client.rb

```diff
@@ -286,9 +286,11 @@ def request(http_method: :get, path:'/', body: nil, query: {}, headers: {}, resp
 
       request_headers = request_headers(headers, auth)
 
-      body_content = body.nil? ? '' : encode_body(body, request_headers[CONTENT_TYPE])
-      if http_method == :get
-        request_headers.delete('Content-Type')
+      if body.nil?
+        body_content = ''
+        request_headers.delete(CONTENT_TYPE)
+      else
+        body_content =  encode_body(body, request_headers[CONTENT_TYPE])
       end
       request_headers.merge!('Content-Length' => body_content.bytesize)
```

Comment:
Thanks for the tip, changed implementation to set `Content-Length` only if body is present